### PR TITLE
WIP: update tests

### LINKED ISSUE #119: make check fails on OSX
On Emacs 24.5 and 25.0.95:
```
🌲  emacs -Q --batch --load test/el-test-runner.el
Loading /Users/cat/Documents/clones/twittering-mode/test/test-twittering-mode.el (source)...
Mark set
Loading international/kinsoku...
Mark set
Mark set
Use Proxy:on
Use Proxy:off
Using a proxy for http...
Mark set
Mark set
Buffer is read-only: #<buffer *test-result*>
```
Exit code is `255`.

Is there a way to get more debugging output for the tests?

### PR DESCRIPTION
**Pull request by dunn**
I marked this as Work In Progress since I don't know the code well enough to be sure I didn't break anything.  But now I can run `make check`, so this fixes #119.

### CODE REVIEW

File: test/el-test-runner.el [hunk 1]
```diff
@@ -1,72 +1,55 @@
...
+;;; Code:
+(require 'test)
+(require 'twittering-mode)
 
-(require 'twittering-mode (expand-file-name "../twittering-mode.el"
-					    twittering-test-dir))
-(require 'test (expand-file-name "../test.el"
-				 twittering-test-dir))
+(setq twittering-test-dir (file-name-directory
+                             (or load-file-name buffer-file-name)))
```
Comment (dunn) [inline test/el-test-runner.el]: Falling back to `buffer-file-name` lets us run this interactively.

File: test/el-test-runner.el [hunk 2]
```diff
@@ -1,72 +1,55 @@
...
   (interactive)
-
   (dolist (file-name (directory-files twittering-test-dir))
     (when (string-match "^test-" file-name)
       (let ((file-name (expand-file-name file-name twittering-test-dir)))
 	(load file-name))))
 
   (test-run-all-cases)
-  
-  (let* ((fail-count (cdr (assoc 'fail test-last-summary)))
```
Comment (dunn) [inline test/el-test-runner.el]: `test-last-summary` was removed from test.el in 0.9.1.

File: test/el-test-runner.el [hunk 3]
```diff
@@ -1,72 +1,55 @@
...
-	 (exit-status (if (eq 0 fail-count) 0 1)))
-    (save-excursion
-      (set-buffer "*test-result*")
-      (end-of-buffer)
-      (search-backward-regexp "" nil t)
-      (beginning-of-buffer)
-      (while (search-forward-regexp "\\b\\([0-9]+\\) pass\\b" nil t)
```
Comment (dunn) [inline test/el-test-runner.el]: At some point test.el changed `*test-result*` to be a read-only buffer, so none of this code seems to work anymore.

File: test/test-twittering-mode.el [hunk 4]
```diff
@@ -78,16 +78,17 @@
...
-				    emacs-minor-version
-				    twittering-mode-version)
-    (twittering-user-agent))
-  (setq twittering-user-agent-function
-	(lambda () "foo user agent"))
-  (test-assert-string-equal "foo user agent"
-    (twittering-user-agent))
-  )
+  (progn
+    (setq twittering-user-agent-function 'twittering-user-agent-default-function)
```
Comment (dunn) [inline test/test-twittering-mode.el]: From what I can tell the tests aren't always run in the exact order that the forms are given, so `(setq twittering-user-agent-function (lambda () "foo user agent"))` was being evaluated before the 
```
(test-assert-string-equal (format "Emacs/%d.%d Twittering-mode/%s" [...]
``` 
test, causing it to fail.

**Comment by tarsius** [comment]
Do you need anything from me before you can merge this?

**Comment by dunn** [comment]
I don't have commit rights; I'm waiting to hear from @cvmat

**Comment by tarsius** [comment]
Ah! Sorry for the noise.

**Pull request closed without merging**